src/main/scala/edu/ie3/mobsim/utils/IoUtils.scala

```diff
@@ -136,23 +137,47 @@ final case class IoUtils private (
     *
     * @param pois
     *   POIs to write
+    * @param evcsDirectHomePoiMapping
+    *   * POIs of Homes that got directly mapped
     */
   def writePois(
-      pois: Map[CategoricalLocationDictionary.Value, Set[PointOfInterest]]
+      pois: Map[CategoricalLocationDictionary.Value, Set[PointOfInterest]],
+      evcsDirectHomePoiMapping: Map[UUID, UUID],
   ): Unit = pois.foreach { case (poiType, typePois) =>
     typePois.foreach { poi =>
       /* Get all charging stations nearby */
-      poi.nearestChargingStations.foreach { case (evcsUuid, distance) =>
+
+      val nearestStations = poi.nearestChargingStations
+      // Write POI data even if there are no nearby charging stations
+      if (nearestStations.isEmpty) {
+
         val fieldData = Map(
-          "uuid" -> UUID.randomUUID().toString,
+          "uuid" -> poi.uuid.toString,
           "id" -> poi.id,
           "type" -> poiType.toString,
           "size" -> poi.size.toString,
-          "evcs" -> evcsUuid.toString,
-          "distance" -> distance.toKilometers.toString,
+          "evcs" -> evcsDirectHomePoiMapping
+            .get(poi.uuid)
+            .orElse(None)
+            .toString,
```

Comment:
Since `None.toString` is `"None"` and I think the empty String would be better here, this is my suggestion:

```suggestion
            .map(_.toString)
            .getOrElse(""),
```